cmd/juju/application/refresher/refresher.go

```diff
@@ -303,7 +303,7 @@ type charmStoreRefresher struct {
 // Allowed will attempt to check if the charm store is allowed to refresh.
 // Depending on the charm url, will then determine if that's true or not.
 func (r *charmStoreRefresher) Allowed(cfg RefresherConfig) (bool, error) {
-	path, err := charm.EnsureSchema(cfg.CharmRef)
+	path, err := charm.EnsureSchema(cfg.CharmRef, charm.CharmHub)
```

Comment:
should be charm.CharmStore for the charmStoreRefresher

Reply:
Nice catch!